src/pybind/mgr/dashboard/frontend/e2e/cluster/silences.po.ts

```diff
@@ -0,0 +1,122 @@
+      addButton.click().then(() => {
+        const createButton = element(by.cssContainingText('button', 'Create Silence'));
+        // wait for modal container where you select Matcher to dissapear
+        browser.wait(
+          Helper.EC.not(
+            Helper.EC.presenceOf(element(by.cssContainingText('.modal-dialog', 'Matcher')))
+          ),
+          5000
+        );
+        createButton.click().then(() => {
+          // wait for browser to return to silences page and table to be visible
+          browser.wait(
+            Helper.EC.presenceOf(this.getTable().first()),
+            Helper.TIMEOUT,
+            'Timed out waiting for silence creation'
+          );
+
+          // Order silences so most recently updated one is on top
+          element(by.cssContainingText('.datatable-header-cell-label.draggable', 'Updated'))
+            .click()
+            .then(() => {
+              element(
+                by.cssContainingText('.datatable-header-cell-label.draggable', 'Updated')
+              ).click();
+            });
+
+          // Opens new silence's details table by clicking on silence then check "comment"
+          // row of details table for silence for the correct comment to verify its existence
+          this.getFirstTableCellWithText(creator).click();
+          expect(element(by.cssContainingText('.datatable-body-row', 'comment')).getText()).toMatch(
+            comment
+          );
+        });
+      });
+    });
+  }
+
+  // expire is currently limited to expiring the most recently updated silence with the given creator name
+  expire(creator) {
+    this.navigateTo();
+
+    expect(this.getBreadcrumbText()).toEqual('Silences');
+
+    // Order silences so most recently updated one is on top
+    element(by.cssContainingText('.datatable-header-cell-label.draggable', 'Updated'))
+      .click()
+      .then(() => {
+        element(by.cssContainingText('.datatable-header-cell-label.draggable', 'Updated')).click();
+      });
+
+    // Opens new silence's details table by clicking on silence then check "comment"
+    // row of details table for silence for the correct comment to verify its existence
+    this.getFirstTableCellWithText(creator).click();
+
+    expect(element(by.cssContainingText('.datatable-body-row', 'state')).getText()).toMatch(
+      'active',
+      'Error: Silence already expired'
+    );
+
+    $('.table-actions button.dropdown-toggle').click(); // click toggle menu
+    $('li.expire a').click(); // click expire
+    browser.wait(Helper.EC.visibilityOf($('.modal-title.float-left')), 10000).then(() => {
+      browser.wait(Helper.EC.visibilityOf($('.custom-control-label')), 5000);
+      $('.custom-control-label').click();
+      element(by.cssContainingText('button', 'Expire Silence'))
+        .click()
+        .then(() => {
+          browser.wait(
+            Helper.EC.not(
+              Helper.EC.presenceOf(element(by.cssContainingText('.modal-dialog', 'Expire Silence')))
```

Comment:
I think it's more clean, when possible, to use 1 EC instead of 2.
```suggestion
              Helper.EC.stalenessOf(element(by.cssContainingText('.modal-dialog', 'Expire Silence')))
```